lib/min_heap.rb

```diff
@@ -14,18 +14,22 @@ def initialize
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(log n)
+  # Space Complexity: O(n)
   def add(key, value = key)
```

Comment:
👍 Why O(n) for space complexity, you add 1 element with this method.

Reply:
I thought n is the number of elements we're adding, not the size of the heap. So by adding 1 heap node we also adding 1 element to the array, that's why O(n).